app/models/service_order.rb

```diff
@@ -4,12 +4,31 @@ class ServiceOrder < ActiveRecord::Base
   has_many :miq_requests, :dependent => :nullify
 
   validates :state, :inclusion => {:in => %w(wish cart ordered)}
-  validates :name, :presence => true
   validates :state, :presence => true
+  validates :name, :presence => true, :on => :update
 
-  before_create :assign_user_name
+  before_create :assign_user_name, :assign_tenant
```

Comment:
@tinaafitz The tests are failing because the user does not get passed through, but they are setting `User.current_user/User.current_tenant` which we can reference here.

I am thinking we change `assign_user_name` to `assign_user` and have it populate `user` and `tenant` for the service order if user is not provided along with setting the user_name field.

Something like this:
```ruby
def assign_user
  self.user ||= begin
    self.tenant = User.current_tenant
    User.current_user
  end
  self.user_name = user.try(:name)
end
```